src/backend/joanie/core/models/courses.py

```diff
@@ -357,15 +357,16 @@ def state(self):
         """
         The state of the course carrying information on what to display on a course glimpse.
 
-        The game is to find the highest priority state for this course among its course runs.
+        The game is to find the highest priority state for this course among
+        its course runs and its products.
         """
-        # The default state is for a course that has no course runs
+        # The default state is for a course that has no course runs or products
         best_state = CourseState(CourseState.TO_BE_SCHEDULED)
+        course_runs = self.course_runs.all()
+        products = self.products.all()
```

Comment:
I think you should instead craft a query on course runs. Something like:
CourseRun.objects.filter(Q(course=self) | Q(products=...))

Reply:
Indeed there is a pain point here, I can try to write only one query.

Reply:
Even if I succeeded to write this query, I think the processed state will not be relevant. Indeed if I shuffle `course.course_runs` and `course.products.target_course_runs`, I will get a state which will not representative of the reality.